aspnet/fundamentals/servers/sample/ServersDemo/src/ServersDemo/Program.cs

```diff
@@ -1,43 +1,93 @@
 ﻿using System;
-using System.Threading.Tasks;
-using Microsoft.AspNet.Hosting;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
-using Microsoft.AspNet.Builder;
-using Microsoft.Extensions.Logging;
-using Microsoft.AspNet.Server.Kestrel;
+using Microsoft.Net.Http.Server;
 
 namespace ServersDemo
 {
     /// <summary>
-    /// This demonstrates how the application can be launched in a console application. 
-    /// Executing the "dnx run" command in the application folder will run this app.
+    /// Executing the "dotnet run --server [Kestrel|WebListener]" command in the application folder will run this app.
+    /// You can also run "dotnet publish" and specify the publish directory as the physical path of a site in IIS Manager.
     /// </summary>
     public class Program
     {
-        private readonly IServiceProvider _serviceProvider;
+        public static string Server;
 
-        public Program(IServiceProvider serviceProvider)
+        public static int Main(string[] args)
         {
-            _serviceProvider = serviceProvider;
-        }
+            // Add command line configuration source to read command line parameters.
+            var config = new ConfigurationBuilder()
+                .AddCommandLine(args)
+                .Build();
 
-        public Task<int> Main(string[] args)
-        {
-            //Add command line configuration source to read command line parameters.
-            var builder = new ConfigurationBuilder();
-            builder.AddCommandLine(args);
-            var config = builder.Build();
-
-            using (new WebHostBuilder(config)
-                .UseServer("Microsoft.AspNet.Server.Kestrel")
-                .Build()
-                .Start())
+            Server = config["server"] ?? "Kestrel";
+
+            var builder = new WebHostBuilder()
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .UseStartup<Startup>();
+
+            // The default url is http://localhost:5000 if none is specified.
```

Comment:
This sample seems to say developers should define their own command line parameter and call UseUrls. They'd be better off using the existing parameter. If you just want to demonstrate UseUrls don't mix it with Configuration.